site/app/templates/grading/electronic/Details.twig

```diff
@@ -697,5 +708,82 @@
         }
         location.reload();
     }
+
+    //function to count how many submissions are being regraded so that a confirm message can be displayed if necessary
+    function ajaxCountRegrade(choice,e,confirmMessage) {
+        const submit_url = buildCourseUrl(['gradeable', "{{ gradeable.getId }}", 'regrade', 'count']);
+        let formData = new FormData();
+        formData.append('csrf_token', csrfToken);
+        formData.append('gradeable_id', "{{ gradeable.getId }}");
+        //if regrading all versions
+        if (choice) {
+            formData.append('regrade_all', 'true');
+        }
+        else {
+            formData.append('regrade_all', 'false');
+        }
+        $.ajax({
+            url: submit_url,
+            data: formData,
+            processData: false,
+            type: 'POST',
+            headers : {
+                Accept: "application/json"
+            },
+            contentType: false,
+            success: function(data) {
+                data = JSON.parse(data);
+                console.log(data);
+                if (data['status'] === 'success') {
+                    confirmMessage(e, data['data'], choice);
+                }
+                else {
+                    if (data['message'] === "You do not have access to that page.") {
+                        window.location.href = '';
+                    }
+                    else {
+                        alert("ERROR! Please contact administrator with following error:\n\n" + data['message']);
+                    }
+                }
+            }
+        });
+    }
+    function confirmMessage(e,count, choice) {
+        if(count > 50) {
+            e.preventDefault();
+            if (window.confirm("Are you sure you want to add " + count + " submissions to the queue ?")) {
+                //if regrading all versions
+                if (choice) {
+                    handleRegrade(0, csrfToken, "{{ gradeable.getId }}", '', false, false, false, true)
+                }
+                else {
+                    handleRegrade(0, csrfToken, "{{ gradeable.getId }}", '', false, false, true)
+                }
+            }
+        }
+        else {
+            //if regrading all versions
+            if (choice) {
+                handleRegrade(0, csrfToken, "{{ gradeable.getId }}", '', false, false, false, true)
+            }
+            else {
+                handleRegrade(0, csrfToken, "{{ gradeable.getId }}", '', false, false, true)
+            }
+        }
+    }
+
+    //when either regrade buttons are clicked, make an ajax call to count how many submissions will be regraded
+    $(function() {
+        $('.regrade-btn').click(function(e) {
+            ajaxCountRegrade(0,e,confirmMessage);
+         });
+    });
+
+    $(function() {
+        $('.regrade-btn-all').click(function(e) {
+            ajaxCountRegrade(1,e,confirmMessage);
+        });
+    });
```

Comment:
Why bind this click handler with JS versus just adding the click handler directly to the `<button>` and `<a>` tags directly?

Reply:
I was having trouble passing the event as a parameter like that but I fixed it.